aten/src/ATen/native/Distance.cpp

```diff
@@ -25,6 +26,33 @@ Tensor pdist(const Tensor& self, const double p) {
   return at::_pdist_forward(self.contiguous(), p);
 }
 
+Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
+    AT_CHECK(x1.dim() == 2, "cdist only supports 2D tensors, X1 got: ", x1.dim(), "D");
+    AT_CHECK(x1.is_contiguous(), "cdist requires X1 to be contiguous");
```

Comment:
can we check non contiguous inputs?  It should not be too difficult.

Reply:
Prior to call kernel Tensor will be turned to contiguous